components/scroll-wrapper/scroll-wrapper.js

```diff
@@ -0,0 +1,214 @@
+				border: 1px solid var(--d2l-color-mica);
+				border-radius: 50%;
+				box-shadow: 0 2px 12px 0 rgba(0, 0, 0, 0.15);
+				box-sizing: content-box;
+				cursor: pointer;
+				display: inline-block;
+				font-family: inherit;
+				font-size: inherit;
+				height: 18px;
+				line-height: 0;
+				margin: 0;
+				padding: 10px;
+				position: absolute;
+				text-decoration: none;
+				top: 4px;
+				width: 18px;
+			}
+			.d2l-scroll-wrapper-button::-moz-focus-inner {
+				border: 0;
+			}
+			.d2l-scroll-wrapper-button:hover,
+			.d2l-scroll-wrapper-button:focus {
+				background-color: var(--d2l-color-sylvite);
+				outline-style: none;
+			}
+			.d2l-scroll-wrapper-button.focus-visible {
+				box-shadow: 0 0 0 2px #ffffff, 0 0 0 4px var(--d2l-color-celestine), 0 2px 12px 0 rgba(0, 0, 0, 0.15);
+			}
+			:host([scrollbar-right]) .d2l-scroll-wrapper-button-right {
+				display: none;
+			}
+			:host([scrollbar-left]) .d2l-scroll-wrapper-button-left {
+				display: none;
+			}
+
+			/* hide wrapper visuals from print view */
+			@media print {
+				.d2l-scroll-wrapper-actions {
+					display: none;
+				}
+				:host([h-scrollbar][show-actions]) .d2l-scroll-wrapper-container {
+					border-left: none;
+					border-right: none;
+				}
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.showActions = false;
+		this._container = null;
+		this._hScrollbar = true;
+		this._scrollbarLeft = false;
+		this._scrollbarRight = false;
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		this._container = this.shadowRoot.querySelector('.d2l-scroll-wrapper-container');
+		new ResizeObserver(() => requestAnimationFrame(() => this.checkScrollbar())).observe(this._container);
+	}
+
+	render() {
+		const actions = this.showActions ? html`
+			<div class="d2l-scroll-wrapper-actions">
+				<button class="d2l-scroll-wrapper-button d2l-scroll-wrapper-button-left" @click="${this._scrollLeft}" aria-hidden="true">
+					<d2l-icon icon="tier1:chevron-left"></d2l-icon>
+				</button>
+				<button class="d2l-scroll-wrapper-button d2l-scroll-wrapper-button-right" @click="${this._scrollRight}" aria-hidden="true">
+					<d2l-icon icon="tier1:chevron-right"></d2l-icon>
+				</button>
+			</div>` : null;
+		return html`
+			${actions}
+			<div class="d2l-scroll-wrapper-container" @scroll="${this._checkScrollThresholds}"><slot></slot></div>
+		`;
+	}
+
+	checkScrollbar() {
+		if (!this._container) return;
+		const hScrollbar = Math.abs(this._container.offsetWidth - this._container.scrollWidth);
+		this._hScrollbar = (hScrollbar > 0);
+		this._checkScrollThresholds();
+	}
+
+	scrollDistance(distance, smooth) {
+		if (!this._container) return;
+		if (this._dir === 'rtl') distance = distance * RTL_MULTIPLIER;
+		if (this._container.scrollBy) {
+			this._container.scrollBy({ left: distance, behavior: smooth ? 'smooth' : 'auto' });
+		} else {
+			// legacy-Edge doesn't support scrollBy
+			this._container.scrollLeft = distance;
```

Comment:
Been a while since I was in scrolling land and can't easily checkout the branch at the moment - does this scroll smooth, or do we need `scroll-behavior: smooth;` in the css?

Reply:
It does not, and I didn't care to spend more time / add more hacks for legacy-Edge since it's going away possibly before this even gets released.

Reply:
Fair enough!